libraries/AP_HAL_PX4/UARTDriver.cpp

```diff
@@ -274,7 +275,10 @@ size_t PX4UARTDriver::write(uint8_t c)
         }
         hal.scheduler->delay(1);
     }
-    return _writebuf.write(&c, 1);
+	_write_sem.take(10);
```

Comment:
@Oxinarf Well without a semaphore writing to a TeraRanger Tower doesn't work. Makes me wonder if it even works for the LightWares driver. Maybe the solution then is to add one or more parameters to the ctor of the UART_Driver class specifying usage of semaphores and the timeout time. Or should I just pass 0 timeout?

@WickedShell Allright I'll look into it.